src/dmd/backend/cgcod.d

```diff
@@ -516,6 +517,9 @@ tryagain:
     }
     else
     {
+        __gshared Barray!ubyte disasmBuf;
```

Comment:
Avoid global variables

Reply:
It's done to avoid constantly allocating and reallocating the memory for it.

Reply:
Then let the caller initialize it

Reply:
It's have to go back multiple levels.

Reply:
I thought you wanted to move dmd in a `@safe` and `pure` direction. I understand legacy code using `__gshared`, but why write new code using it?

Reply:
The idea is to get things to work with minimal changes to the code base. Refactorings come later. Figuring out how to remove globals is a case-by-case thing, and unnecessarily adds complexity to this PR.